src/app/components/keyboard/keyboard.component.html

```diff
@@ -1,31 +1,31 @@
 <div class="keyboard">
   <div class="container w-50">
-    <div class="row">
+    <div class="row" *ngIf="firstKeyboardRow">
```

Comment:
```suggestion
    <div class="row">
```
Maybe the ngIfs are not necessary, try it and remove it from the code

Reply:
right, it was not necessary